pkg/controller/secret/secret_controller.go

```diff
@@ -407,6 +407,17 @@ func createAlertManagerConfig(pagerdutyRoutingKey, watchdogURL, consoleUrl strin
 					"alertname": "KubeDeploymentReplicasMismatch",
 				},
 			},
+			{
+				Equal: []string{
+					"dummylabel",
```

Comment:
NB this label obviously won't match and that's both ok and expected. When a label is missing (or empty) on both source and target, the rule will apply (see: [docs](https://www.prometheus.io/docs/alerting/latest/configuration/#inhibit_rule) ).

If there wasn't a label here, the tests exploded spectacularly, so I figured a label that would never match is the next best thing.

Reply:
Is it possible to get a code comment describing this? I figure this is something we will be scratching our heads about in the future